libs/src/api/rewards.js

```diff
@@ -461,12 +467,13 @@ class Rewards extends Base {
     endpoints,
     numAttestations = 3
   }) {
-    if (!endpoints) {
-      endpoints = await this.discoveryProvider.serviceSelector.findAll()
+    let attestEndpoints
+    if (endpoints) {
+      attestEndpoints = sampleSize(endpoints, numAttestations)
+    } else {
+      attestEndpoints = await this.ServiceProvider.getUniquelyOwnedDiscoveryNodes(numAttestations)
     }
-    const attestEndpoints = shuffle(
-      endpoints
-    ).filter(s => s !== senderEndpoint).slice(0, numAttestations)
+
```

Comment:
might be worth pulling this out to a private fn in rewards

Reply:
b/c the two functions use different vars for quorum size (quorumSize and numAttestations), i think broken out as a separate helper is kinda confusing. At first I wanted to replace the numAttestations with quorumSize, but that does have downstream impact I didn't really want to deal with now. lmk if you feel strongly